controllers/component_build_controller_simple_build.go

```diff
@@ -136,14 +135,18 @@ func (r *ComponentBuildReconciler) getBuildGitInfo(ctx context.Context, componen
 	}
 	if !isPublic {
 		// Try to find git secret in case private git repository is used
-		gitSecretName, err = r.findGitSecretName(ctx, component)
-		if err != nil {
-			log.Error(err, "failed to find git secret name")
-			return nil, err
-		}
+		gitSecretName = component.Spec.Secret
 		if gitSecretName == "" {
-			// The repository is private and no git secret provided
-			return nil, boerrors.NewBuildOpError(boerrors.EComponentGitSecretMissing, nil)
+			// The repository is private and no git secret specified
+			return nil, boerrors.NewBuildOpError(boerrors.EComponentGitSecretNotSpecified, nil)
```

Comment:
What would be the difference between `boerrors.EComponentGitSecretMissing`  and `boerrors.EComponentGitSecretNotSpecified` here?

Reply:
`NotSpecified` happens when the secret name was not given, where as `Missing` when secret name is specified, but the secret itself does not exist.

Reply:
I see the description in `pkg/boerrors/perror.go` but I am not sure I completely understand

Reply:
The diagram shows the process is:

UI HAC creates RemoteSecret -> RemoteSecret creates the Secret -> UI HAC update the Secret name to component

Should we treat the this error as transient error and emit Event as well?